lib/relation-definition.js

```diff
@@ -532,37 +532,85 @@ function lookupModel(models, modelName) {
   }
 }
 
-function lookupModelTo(modelFrom, modelTo, params, singularize) {
-  if ('string' === typeof modelTo) {
+/*
+ * @return {Object} modelFrom Instance of the 'from' model
+ * @param {Object|String} modelToRef Reference to Model object to which you are
+ *  creating the relation: model instance, model name, or name of relation to model.
+ * @param {Object} params relation params
+ * @param {Boolean} singularize Whether the modelToRef should be singularized when
+ *  looking-up modelTo
+ * @return {Object} modelTo Instance of the 'to' model
+ */
+function lookupModelTo(modelFrom, modelToRef, params, singularize) {
+  var modelTo;
+
+  if ('string' === typeof modelToRef) {
     var modelToName;
-    params.as = params.as || modelTo;
-    modelTo = params.model || modelTo;
-    if (typeof modelTo === 'string') {
-      modelToName = (singularize ? i8n.singularize(modelTo) : modelTo).toLowerCase();
-      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName) || modelTo;
-    }
+    modelTo = params.model || modelToRef; // modelToRef is modelTo name
+    params.as = params.as || modelToRef;  // modelToRef is relationTo name
+
     if (typeof modelTo === 'string') {
-      modelToName = (singularize ? i8n.singularize(params.as) : params.as).toLowerCase();
-      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName) || modelTo;
+      // lookup modelTo by name
+      modelToName = modelTo;
+      modelToName = (singularize ? i8n.singularize(modelToName) : modelToName).toLowerCase();
+      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName);
     }
-    if (typeof modelTo !== 'function') {
-      throw new Error(g.f('Could not find "%s" relation for %s', params.as, modelFrom.modelName));
+    if (!modelTo) {
```

Comment:
according to `if (typeof modelTo === 'string') {}` and `modelTo = params.model || modelToRef;`, I don't think `!modelTo` will be true....so this condition won't happen